src/CozyUtils.js

```diff
@@ -153,44 +154,108 @@ class CozyUtils {
    * @returns {object}
    */
   async computeShortcuts(files) {
+    log('info', 'computeShortcuts starts')
     let schoolShortcuts = []
     let favShortcuts = []
     for (const file of files) {
       if (file.hubMetadata.favori) {
         favShortcuts.push({
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
+          // metadata: {
+          //   icon: svgIcon
+          // },
           shouldReplaceFile: true
         })
       } else {
-        schoolShortcuts.push({
+        const appToSave = {
           ...file,
-          tempAppId: file.hubMetadata.idInterne,
+          vendorRef: file.hubMetadata.idInterne,
           filename: `${file.title}.url`,
           filestream: `[InternetShortcut]\nURL=${file.url}`,
           shouldReplaceFile: true
-        })
+        }
+        if (file.icon) {
+          if (file.icon.match('https://')) {
+            appToSave.metadata = {
+              icon: file.icon
+            }
+          }
+        }
+        schoolShortcuts.push(appToSave)
       }
     }
     return { schoolShortcuts, favShortcuts }
   }
 
-  // Waiting for the full svg icons to be handled
-  async computeThumbnails(files) {
-    let thumbnailsSource = []
-    for (const file of files) {
-      if (file.thumbnail === undefined) {
+  async findShortcuts() {
+    log('info', 'Getting in findShortcuts')
+    // Here we're looking for the shortcuts created by toutatice konnector
+    // We know their path is "/Settings/Home", so there is no need to query the path
+    try {
+      const query = Q(DOCTYPE_FILES).partialIndex({
+        type: 'file',
+        class: 'shortcut',
+        'cozyMetadata.createdByApp': 'toutatice',
+        trashed: false
+      })
+      const existingShortcuts = await this.client.queryAll(query)
+      return existingShortcuts
+    } catch (err) {
+      throw new Error(err)
+    }
+  }
+
+  async synchronizeShortcuts(foundShortcuts, computedShortcuts, favFolder) {
+    const favFolderId = favFolder._id
+    let allComputedShortcuts = computedShortcuts.schoolShortcuts.concat(
+      computedShortcuts.favShortcuts
+    )
+    let appsToDelete = []
+    for (let cozyShortcut of foundShortcuts) {
+      const isFavourite = favFolderId === cozyShortcut.dir_id ? true : false
+      if (cozyShortcut.type !== 'file') {
```

Comment:
You can remove this line, there is no way you'll have something else than a file here.